### PR TITLE
Refactor CC installation tests

### PR DESCRIPTION
**Pull request by paolostivanin**
Ticket: https://bugzilla.suse.com/show_bug.cgi?id=1228795 + https://progress.opensuse.org/issues/167659

Related MR: https://gitlab.suse.de/qe-security/osd-sle15-security/-/merge_requests/306

VRs:
- 15-SP7 (x86_64, s390x, aarch64): https://openqa.suse.de/tests/overview?distri=sle&version=15-SP7&build=31.1&groupid=431
- 15-SP6 (x86_64, s390x, aarch64): https://openqa.suse.de/tests/overview?distri=sle&version=15-SP6&build=20241021-1&groupid=431
- 15-SP4 (x86_64, s390x, aarch64): https://openqa.suse.de/tests/overview?distri=sle&version=15-SP4&build=20241021-1&groupid=431
- 15-SP3 (x86_64, s390x, aarch64): https://openqa.suse.de/tests/overview?distri=sle&version=15-SP3&build=20241021-1&groupid=431
- 15-SP5-QU (s390x, x86_64): https://openqa.suse.de/tests/overview?distri=sle&version=15-SP5&build=146.3&groupid=431
- 15-SP4-QU (aarch64, s390x): https://openqa.suse.de/tests/overview?distri=sle&version=15-SP4&build=209.1&groupid=431

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -338,11 +340,11 @@ C<$check_typed_password> will default to C<0>.
 sub unlock_if_encrypted {
     my (%args) = @_;
     $args{check_typed_password} //= 0;
+    my $password = check_var('SYSTEM_ROLE', 'Common_Criteria') ? $security::config::strong_password : $testapi::password;
```
Comment (foursixnine) [inline lib/utils.pm:343]: This fits better in: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/2a0c39849ca58afff6af72ddfb05b886467166e5/lib/main_common.pm#L149

Comment (foursixnine) [reply lib/utils.pm:343]: See also the following message, about documenting the test data https://suse.slack.com/archives/C02CANHLANP/p1728634460708519?thread_ts=1728632342.244079&cid=C02CANHLANP

File: lib/security/config.pm [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+package security::config;
+
+use strict;
+use warnings;
+
+use base 'Exporter';
+
+our @EXPORT = qw(strong_password);
+
+our $strong_password = "not-so-s3cr3t_BUT_VERY_STRONG.";
```
Comment (foursixnine) [inline lib/security/config.pm:18]: If this is going to be hardcoded, please do make sure that there is some POD documentation about it that can be shown at https://os-autoinst.github.io/os-autoinst-distri-opensuse/registration.html

```suggestion
our $strong_password = "not-so-s3cr3t_BUT_VERY_STRONG.";
```

**Review by foursixnine — CHANGES_REQUESTED** [review]

File: lib/utils.pm (hunk 1 above)
Comment (paolostivanin) [reply lib/utils.pm:343]: thanks, I'll do that!

File: lib/security/config.pm (hunk 2 above)
Comment (paolostivanin) [reply lib/security/config.pm:18]: ok, will do that!

File: lib/utils.pm (hunk 1 above)
Comment (paolostivanin) [reply lib/utils.pm:343]: actually, that password is just for disk encryption and not for the user itself.

File: lib/security_boot_utils.pm [hunk 3]
```diff
@@ -25,9 +27,8 @@ our @EXPORT = qw(
 
 sub boot_has_no_video {
     my $is_encrypted = check_var('FULL_LVM_ENCRYPT', '1') || check_var('ENCRYPT', '1');
-    my $is_qr = check_var('FLAVOR', 'Online-QR') || check_var('FLAVOR', 'Full-QR');
-    my $is_arch = is_aarch64() || is_s390x();
-    return ($is_encrypted && $is_qr && $is_arch);
+    my $is_arch = is_aarch64();
+    return ($is_encrypted && $is_arch);
```
Comment (ilmanzo) [inline lib/security_boot_utils.pm:31]: I'd remove the variable and shorten to 
```Perl
return ($is_encrypted && $is_aarch64());
```

**Pull request merged**